nemo/utils/secure.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import safetensors
+import safetensors.torch as storch
+import torch
+
+from nemo.utils import logging
+
+SAFE_EXTENSION = '.safetensor'
+
+
+def torch_load(filename: str, map_location: str = 'cpu', safe: bool = False):
+    """
+    load PyTorch files, for backwards compatability it loads insecurely
+    by default because files have to be saved securely to work perfectly,
+    however when loading untrusted files, safe should be set to True.
+
+    Args:
+        filename: the file to load
+        map_location: the device to load the file onto
+        safe: whether to load the file with safetensors
+    Returns:
+        the loaded tensors
+    """
+    try:
+        return storch.load_file(filename + SAFE_EXTENSION, device=map_location)
+    except FileNotFoundError as e:
+        if safe:
+            raise e
+        else:
+            logging.info(e)
```

Comment:
My opinion is that this is going to create a lot of confusingly noise. Basically every time that someone does an unsafe load on a model that wasn't saved with the safe=True option (which is frequent because it's the default), they will get an unhelpful log message of a file not found error, even though the application is otherwise working correctly.